homeassistant/components/media_player/__init__.py

```diff
@@ -297,6 +304,12 @@ async def async_setup(hass, config):
         "async_media_seek",
         [SUPPORT_SEEK],
     )
+    component.async_register_entity_service(
+        SERVICE_JOIN,
+        {vol.Required(ATTR_GROUP_MEMBERS): list},
```

Comment:
Should there be a way to define a leader ?

Reply:
@balloob Thank you very much for your input. I was initially thinking about explicitly specifying the leader as well. However @andrewsayre suggested to leave this out (https://github.com/home-assistant/core/pull/32568#pullrequestreview-454203730).

Specifying a leader could make sense for some integrations (HEOS for example), as the leader speaker is static there. However some multiroom audio systems (such as Sonos) have the concept of taking over leadership dynamically, if the leader speaker is removed from an existing group. There *might* also be other multiroom audio systems out there which don't have a leader concept at all, but I am not sure about this.

So the idea of @andrewsayre was to hide the leader concept entirely and let the integration handle the details.

But yeah, in the end we have multiple options here:

1. Hide the leader concept from the service calls and leave it up to the integration

2. Require an explicit leader in the service call

3. Make the leader an optional argument in the service call

I am fine with either approach. Currently I chose **1.** but I'll happily implement one of the other ones. I must admit that I am a little stuck right now because I don't know where I can get proper feedback for such core decisions. The arch issue (home-assistant/architecture/issues/364) does not contain any comments regarding this. How do we want to handle this?

Reply:
Hiding the leader concept seems ok 👍

Reply:
OK, perfect.